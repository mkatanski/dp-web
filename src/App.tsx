import React, { Suspense } from "react";
import { BrowserRouter as Router } from "react-router-dom";

import Routes from "routes";
import { LoadingScreen } from "components/fallbacks/LoadingScreen";

import { MainStyle } from "styles/globalStyles";
import { ThemeProvider } from "lib/theme";
import { ThemeProvider as MaterialThemeProvider } from "@material-ui/styles";
import { StylesProvider } from "@material-ui/core/styles";

import { DefaultLightTheme } from "styles/theme/light";

import styled from "styled-components";

const Container = styled.div`
  min-height: 100vh;
  height: 100vh;
  padding: 0;
  margin: 0;
  display: flex;
  align-items: center;
  justify-content: center;
`;

const App = () => {
  return (
    <div className="App">
      <StylesProvider injectFirst>
        <ThemeProvider theme={DefaultLightTheme}>
          <MaterialThemeProvider theme={DefaultLightTheme}>
            <MainStyle />
            <Router>
              <Suspense
                fallback={
                  <Container>
                    <LoadingScreen />
                  </Container>
                }
              >
                <Routes />
              </Suspense>
            </Router>
          </MaterialThemeProvider>
        </ThemeProvider>
      </StylesProvider>
    </div>
  );
};

export default App;
