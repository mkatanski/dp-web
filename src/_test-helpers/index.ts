export * from "./withResponsiveContext";
export * from "./withThemeProvider";
