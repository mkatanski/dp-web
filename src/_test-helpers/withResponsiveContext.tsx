import React from "react";
import { Context as ResponsiveContext } from "react-responsive";
import { Breakpoints } from "config/breakpoints";

export const withMobileSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.mobile }}>
    {Component}
  </ResponsiveContext.Provider>
);

export const withTabletSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.tablet }}>
    {Component}
  </ResponsiveContext.Provider>
);

export const withDesktopSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.desktop }}>
    {Component}
  </ResponsiveContext.Provider>
);

export const withLargeDesktopSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.largeDesktop }}>
    {Component}
  </ResponsiveContext.Provider>
);
