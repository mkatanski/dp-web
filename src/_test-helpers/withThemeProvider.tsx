import React from "react";
import { ThemeProvider as DefaultThemeProvider } from "lib/theme/ThemeProvider";
import { StyledTheme, Theme } from "styles/theme/_types";
import { createMuiTheme } from "@material-ui/core";

const testTheme: StyledTheme = {
  primary: "primary",
  secondary: "secondary",
  textPrimary: "textPrimary",
  textSecondary: "textSecondary",
  textInverse: "textInverse"
};

const muiTheme = createMuiTheme({
  direction: "ltr"
});

const theme: Theme = Object.assign(testTheme, muiTheme);

export const withThemeProvider = (Component: React.ReactNode) => (
  <DefaultThemeProvider theme={theme}>{Component}</DefaultThemeProvider>
);
