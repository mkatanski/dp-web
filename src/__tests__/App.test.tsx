/* eslint-disable react/display-name */
import React from "react";
import TestRenderer from "react-test-renderer";
import "jest-styled-components";
import App from "App";

import Routes from "../routes";

import { withThemeProvider } from "_test-helpers";

jest.mock("routes", () => jest.fn());

jest.mock("react-router-dom", () => ({
  BrowserRouter: (props: React.PropsWithChildren<{}>) => (
    <div id="BrowserRouter">{props.children}</div>
  )
}));

jest.mock("@material-ui/styles", () => ({
  ...require.requireActual("@material-ui/styles"),
  ThemeProvider: (props: React.PropsWithChildren<{}>) => (
    <div id="MUI.ThemeProvider">{props.children}</div>
  )
}));

jest.mock("components/fallbacks/LoadingScreen", () => ({
  LoadingScreen: jest.fn(() => {
    return <div id="LoadingScreen" />;
  })
}));

describe("App", () => {
  it("should render App component", async () => {
    (Routes as jest.Mock).mockImplementationOnce(() => <div>Routes</div>);

    const tree = TestRenderer.create(<App />);
    expect(tree).toMatchSnapshot();
  });

  it("should render a loading screen as a fallback", async () => {
    (Routes as jest.Mock).mockImplementationOnce(() => {
      throw new Promise(r => {
        r();
      });
    });
    const tree = TestRenderer.create(withThemeProvider(<App />));
    expect(tree).toMatchSnapshot();
  });
});
