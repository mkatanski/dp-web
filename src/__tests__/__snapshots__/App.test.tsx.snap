// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render App component 1`] = `
<div
  className="App"
>
  <div
    id="MUI.ThemeProvider"
  >
    <div
      id="BrowserRouter"
    >
      <div>
        Routes
      </div>
    </div>
  </div>
</div>
`;

exports[`App should render a loading screen as a fallback 1`] = `
.c0 {
  min-height: 100vh;
  height: 100vh;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="App"
>
  <div
    id="MUI.ThemeProvider"
  >
    <div
      id="BrowserRouter"
    >
      <div
        className="c0"
      >
        <div
          id="LoadingScreen"
        />
      </div>
    </div>
  </div>
</div>
`;
